kedro/io/data_catalog.py

```diff
@@ -713,6 +725,9 @@ def add_feed_dict(self, feed_dict: dict[str, Any], replace: bool = False) -> Non
 
             self.add(dataset_name, dataset, replace)
 
+    # Alias for add_feed_dict method
+    from_dict = add_feed_dict
```

Comment:
Pedantic nitpick, but when I read `from_dict` I assume it's a classmethod, like

```python
class DataCatalog:
  @classmethod
  def from_dict(cls, data_dict: dict[str, AbstractDataset]):
    ...
    return cls
```

So not sure if I'd add this alias.

(I know this comes from https://github.com/kedro-org/kedro/issues/3612#issuecomment-2034961020 and that I asserted, but my response there is incomplete)

Reply:
This is actually quite a significant change. I think we should discuss first what a good alternative is, before just adding an alias.

Reply:
This has always been an undiscovered part of the API, I agree with @merelcht that we should think really hard / research what a better name should be.

Reply:
I removed an alias until we are sure about adding it and agree on its name.

To @astrojuanlu's point, if we decide to keep an alias, we can use the `add_from_dict` name instead.

Or, as an alternative, we can add a new implementation (`add_from_dict` / `add_datasets`) and use it inside `add_feed_dict` with a deprecation warning.